lib/atlas/scaler.rb

```diff
@@ -49,12 +55,19 @@ def new_attributes
       }
     end
 
-    def copy_etengine_data_files
-      FileUtils.cp_r(
-        UNSCALED_ETENGINE_DATA_FILES.
-          map    { |subdir| File.join(@base_dataset.dataset_dir, subdir) }.
-          select { |subdir| File.directory?(subdir) },
-        @derived_dataset.dataset_dir)
+    def symlink_etengine_data_files
+      UNSCALED_ETENGINE_DATA_FOLDERS.each do |folder|
+        base     = @base_dataset.dataset_dir.join(folder)
+        new_base = @derived_dataset.dataset_dir.join(folder)
+
+        if File.directory?(base)
+          FileUtils.mkdir_p(new_base)
+
+          Dir[base.join('*')].each do |file|
```

Comment:
This is going to symlink the individual files rather than the directories. That has some advantages (the ability to change a single file without duplicating the directory) but it's not what has been done so far with derived datasets. Do you think it's better to link the files instead of the directories?

A downside is that if we add a new profile to a base dataset, we have to manually symlink it in all the derived datasets.

Reply:
> A downside is that if we add a new profile to a base dataset, we have to manually symlink it in all the derived datasets.

This was already the case with copy/pasting aswell for example `curves/heat*`; people had to manually copy over the `curves` directory because it wasn't in the scale script. I think that having a rake task for this would be wise. Much like `refresh_graph` have a `update_curves` task which updates these curves (so create new symlinks where needed).

Reply:
> This was already the case with copy/pasting aswell for example `curves/heat/*`

Indeed. I'm not debating that though; I dropped the ball by not having "curves/..." symlinked when creating a new derived dataset, sorry. I completely agree it *should* be symlinked automatically when creating a new dataset.

What I'm saying is: should we link the files themselves, or the directories that contain the files?

```
derived/curves/heat/1987/air_temperature.csv -> nl/curves/heat/1987/air_temperature.csv
derived/curves/heat/1987/insulated_household.csv -> nl/curves/heat/1987/insulated_household.csv
derived/curves/heat/1987/non_insulated_household.csv -> nl/curves/heat/1987/non_insulated_household.csv
derived/curves/heat/default/air_temperature.csv -> nl/curves/heat/default/air_temperature.csv
...
```

... or ...


```
derived/curves/heat/1987 -> nl/curves/heat/1987
derived/curves/heat/default -> nl/curves/heat/default
derived/load_profiles -> nl/load_profiles
...
```

I think the first is better only if people regularly want to modify a single file. But I don't think that ever happens. If you know that this is going to be common with future datasets, then I'm completely happy to go ahead with it.

> Much like `refresh_graph`

People are always forgetting to run this. :wink:

Reply:
Okay, I asked the dataset people and it never happens that one of those files is changed or touched 👍